src/App.tsx

```diff
@@ -14,11 +15,166 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Goods</h1>
-    {goodsFromServer.length}
-  </div>
-);
+type State = {
+  goods: string[];
+  visibleList: boolean;
+  isReversed: boolean;
+  sortByAlphabet: boolean;
+  sortByLength: boolean;
+  limit: number;
+};
 
-export default App;
+export class App extends React.Component<{}, State> {
+  state = {
+    goods: goodsFromServer,
+    visibleList: false,
+    isReversed: false,
+    sortByAlphabet: false,
+    sortByLength: false,
+    limit: 1,
+  };
+
+  adderList = () => {
+    this.setState((state) => ({
+      visibleList: !state.visibleList,
+    }));
+  };
+
+  reverseList = () => {
+    this.setState((state) => ({
+      isReversed: !state.isReversed,
+    }));
+  };
```

Comment:
can you replace this.state.goods with reversed copy goods (and you delete this logic from render)? You can do this with all sorted methods